src/lsm/fuzz.zig

```diff
@@ -0,0 +1,459 @@
+
+    try Environment.format(&storage);
+    try Environment.run(&storage, fuzz_ops);
+}
+
+pub fn generate_fuzz_ops(random: std.rand.Random) ![]const FuzzOp {
+    const fuzz_op_count = randomIntExp(usize, random, 1E6);
+    const fuzz_ops = try allocator.alloc(FuzzOp, fuzz_op_count);
+
+    // We're not allowed to go more than Environment.cache_entries_max puts without compacting.
+    var puts_since_compact: usize = 0;
+
+    for (fuzz_ops) |*fuzz_op| {
+        fuzz_op.* = if (puts_since_compact >= Environment.cache_entries_max)
+            // We have to compact before doing any other operations.
+            FuzzOp{ .compact = {} }
+        else
+        // Otherwise pick a random FuzzOp.
+        switch (random.enumValue(std.meta.Tag(FuzzOp))) {
+            .compact => FuzzOp{
+                .compact = {},
+            },
+            .put => FuzzOp{
+                .put = Account{
+                    // Use exponentially distributed ids to increase the chances of overwriting previous values.
+                    .id = randomIntExp(u128, random, 10),
+                    // TODO It would be nice to make timestamp random, but we have to ensure that they match the timestamp of previous Accounts with the same id.
+                    .timestamp = 0,
+                    .user_data = 0,
+                    .reserved = [_]u8{0} ** 48,
+                    .ledger = random.int(u32),
+                    .code = random.int(u16),
+                    .flags = .{
+                        .debits_must_not_exceed_credits = random.boolean(),
+                        .credits_must_not_exceed_debits = random.boolean(),
+                    },
+                    .debits_pending = random.int(u64),
+                    .debits_posted = random.int(u64),
+                    .credits_pending = random.int(u64),
+                    .credits_posted = random.int(u64),
+                },
+            },
+            .get => FuzzOp{
+                .get = .{
+                    .id = randomIntExp(u128, random, 10),
+                },
+            },
+        };
+        switch (fuzz_op.*) {
+            .compact => puts_since_compact = 0,
+            .put => puts_since_compact += 1,
+            .get => {},
+        }
+    }
+
+    return fuzz_ops;
+}
+
+pub fn main() !void {
+    var seed: ?u64 = null;
+
+    var args = std.process.args();
+    {
+        // The first arg is the executable name.
+        const exe = try args.next(allocator).?;
+        defer allocator.free(exe);
+    }
+    while (args.next(allocator)) |arg_or_err| {
+        const arg = try arg_or_err;
+        defer allocator.free(arg);
+        if (std.mem.eql(u8, arg, "--seed")) {
+            const seed_string_or_err = args.next(allocator) orelse
+                std.debug.panic("Expected an argument to --seed", .{});
+            const seed_string = try seed_string_or_err;
+            defer allocator.free(seed_string);
+            if (seed != null)
```

Comment:
We typically use braced for multi-line `if`s.

Reply:
:+1: 

Fixed.